cob_twist_controller/src/inverse_jacobian_calculations/inverse_jacobian_calculation.cpp

```diff
@@ -76,18 +77,18 @@ Eigen::MatrixXd PInvBySVD::calculate(const TwistControllerParams& params,
         for (uint32_t i = 0; i < singularValues.rows()-1; ++i)
         {
             // pow(beta, 2) << pow(lambda, 2)
-            singularValuesInv(i) = singularValues(i) / (pow(singularValues(i), 2) + pow(params.beta, 2));
+            singularValuesInv(i) = singularValues(i) / (pow(singularValues(i), 2) + pow(params.beta, 2)+ lambda(i,i));
         }
         // Formula 20 - additional part - numerical filtering for least singular value m
-        uint32_t m = singularValues.rows();
-        singularValuesInv(m) = singularValues(m) / (pow(singularValues(m), 2) + pow(params.beta, 2) + pow(lambda, 2));
+        //uint32_t m = singularValues.rows();
+        //singularValuesInv(m) = singularValues(m) / (pow(singularValues(m), 2) + pow(params.beta, 2) + pow(lambda, 2));
     }
     else
     {
         // small change to ref: here quadratic damping due to Control of Redundant Robot Manipulators : R.V. Patel, 2005, Springer [Page 13-14]
         for (uint32_t i = 0; i < singularValues.rows(); ++i)
         {
-            double denominator = (singularValues(i) * singularValues(i) + pow(lambda, 2) );
+            double denominator = (singularValues(i) * singularValues(i) + lambda(i,i) );
```

Comment:
Here you can see that a ZeroMatrix with only one entry on the diagonal is different from DiagonalMatrix(lambda), because for all indices i not equal singularValues.rows()-1, this summand will be 0.0